src/shared/varlink.c

```diff
@@ -193,6 +194,8 @@ struct Varlink {
         sd_event_source *time_event_source;
         sd_event_source *quit_event_source;
         sd_event_source *defer_event_source;
+
+        pid_t exec_pid;
```

Comment:
not really necessary: it's our own child, hence we know exactly how long the pid is valid.

There might be value in eventually switching *everything* to PidRef, just as a matter of principle, but we aren't there yet, as safe_fork() will return you a pid_t, not a PidRef as of now.

Hence, there's not much a PidRef would gain us here for now, and would mean we'd have to do even more infrastructure work first, to teach safe_fork() PidRef and so on.

(and if we do that, then your sd-executor PR would probably be fucked up because exec_spawn() would also have to be updated to do PidRef